packages/xrpl/src/models/methods/ledger.ts

```diff
@@ -96,30 +202,57 @@ export interface LedgerBinary
   transactions?: string[]
 }
 
+interface LedgerResponseBase {
+  /** Unique identifying hash of the entire ledger. */
+  ledger_hash: string
+  /** The Ledger Index of this ledger. */
+  ledger_index: number
+  /**
+   * If true, this is a validated ledger version. If omitted or set to false,
+   * this ledger's data is not final.
+   */
+  queue_data?: Array<LedgerQueueData | string>
+  /**
+   * Array of objects describing queued transactions, in the same order as
+   * the queue. If the request specified expand as true, members contain full
+   * representations of the transactions, in either JSON or binary depending
+   * on whether the request specified binary as true.
+   */
+  validated?: boolean
+}
+
+interface LedgerResponseResult extends LedgerResponseBase {
+  /** The complete header data of this {@link Ledger}. */
+  ledger: LedgerBinary
+}
+
+// This is the default
+// expand = false
+// binary = false
+// Matches basic LedgerRequest
 /**
  * Response expected from a {@link LedgerRequest}.
  *
  * @category Responses
  */
 export interface LedgerResponse extends BaseResponse {
-  result: {
-    /** The complete header data of this {@link Ledger}. */
-    ledger: Ledger | LedgerBinary
-    /** Unique identifying hash of the entire ledger. */
-    ledger_hash: string
-    /** The Ledger Index of this ledger. */
-    ledger_index: number
-    /**
-     * If true, this is a validated ledger version. If omitted or set to false,
-     * this ledger's data is not final.
-     */
-    queue_data?: Array<LedgerQueueData | string>
-    /**
-     * Array of objects describing queued transactions, in the same order as
-     * the queue. If the request specified expand as true, members contain full
-     * representations of the transactions, in either JSON or binary depending
-     * on whether the request specified binary as true.
-     */
-    validated?: boolean
-  }
+  result: LedgerResponseResult
+}
+
+interface LedgerResponseExpandedResult extends LedgerResponseBase {
+  /** The complete header data of this {@link Ledger}. */
+  ledger: Ledger
+}
+
+// expand = true
+// accounts OR transactions OR both
+// binary missing altogether
+// matches LedgerRequestExpanded
```

Comment:
These comments seem like they were notes during development, but I could be wrong

Reply:
Fxied